doc/examples/realtime_servo/realtime_servo_tutorial.rst

```diff
@@ -1,181 +1,196 @@
+MoveIt Servo facilitates realtime control of your robot arm.
 
-This tutorial shows how to use MoveIt Servo to send real-time servo commands to a ROS-enabled robot. Some nice features of the servo node are singularity handling and collision checking that prevents the operator from breaking the robot.
+Servo accepts any of the following types of commands:
 
-.. raw:: html
+    1. Individual joint velocities.
+    2. The desired velocity of end-effector.
+    3. The desired pose of end-effector.
 
-    <div style="position: relative; padding-bottom: 5%; height: 0; overflow: hidden; max-width: 100%; height: auto;">
-        <iframe width="700px" height="400px" src="https://www.youtube.com/embed/MF-_XKpGefY" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
-    </div>
+This enables teleoperation via a wide range of input schemes, or for other autonomous software to control the robot - in visual servoing or closed loop position control for instance.
 
 Getting Started
 ---------------
+
 If you haven't already done so, make sure you've completed the steps in :doc:`Getting Started </doc/tutorials/getting_started/getting_started>`.
 
-Launching a Servo Node
-----------------------
-MoveIt Servo can be launched as a "node component" or a standalone node. The launch file, moveit_servo/servo_example.launch.py, launches a standalone node by default but also contains a commented component node. Commands are sent via ROS topics. The commands can come from anywhere, such as a joystick, keyboard, or other controller.
 
-This demo was written for an Xbox 1 controller, but can be easily modified to use any controller compatible with the `Joy package <https://index.ros.org/p/joy/#{DISTRO}>`_ by modifying the `joystick_servo_example.cpp file <https://github.com/ros-planning/moveit2/blob/main/moveit_ros/moveit_servo/src/teleop_demo/joystick_servo_example.cpp>`_
+Design overview
+---------------
+
+Moveit Servo consists of two main parts, the core implementation ``Servo`` which provides a C++ interface, and the ``ServoNode`` which
+wraps the C++ interface and provides a ROS interface.
+
+The configuration of Servo is done through ROS parameters specified in `servo_parameters.yaml <https://github.com/ros-planning/moveit2/blob/main/moveit_ros/moveit_servo/config/servo_parameters.yaml>`_
+
+In addition to the servoing capability, MoveIt Servo has some convenient features such as:
+
+    - Checking for singularities
+    - Checking for collisions
+
+The collision checking is an optional feature that can be disabled using the ``check_collisions`` parameter.
+
+Using the C++ API
+------------------
+The C++ interface allows the user to create and utilize an instance of Servo within their C++ programs.
+This can be beneficial when there is a performance requirement to avoid the overhead of ROS communication infrastucture,
+or when the output generated by Servo needs to be fed into some other controller that does not have a ROS interface.
```

Comment:
```suggestion
This can be beneficial when there is a performance requirement to avoid the overhead of ROS communication infrastucture, or when the output generated by Servo needs to be fed into some other controller that does not have a ROS interface.
```